src/Components/PaperEditor/TranscriptsContainer/TranscriptTabContent/index.js

```diff
@@ -116,49 +110,215 @@ const TranscriptTabContent = (props) => {
     }
   }, [ projectId, transcriptId, firebase.storage, media.ref, url ]);
 
-  const showLabelsReference = () => {};
+  useEffect(() => {
+    const highlightWords = (words) => {
+      const dataParagraphText = words.join(' ');
+
+      const css = words.reduce(
+        (res, word) => {
+          res.paragraphs.push(
+            `.paragraph[data-paragraph-text*="${ dataParagraphText }"] > div > span.words[data-text="${ word }"]`
+          );
+          // Need to add an extra span to search annotation highlights
+          res.search.push(
+            `.paragraph[data-paragraph-text*="${ dataParagraphText }"] > div > span > span.words[data-text="${ word }"]`
+          );
+
+          return res;
+        },
+        { paragraphs: [], search: [] }
+      );
+
+      css.paragraphs = css.paragraphs.join(', ');
+      css.search.join(', ');
+
+      return css;
+    };
 
-  const handleLabelsSearchChange = (selectedOptionLabelSearch) => {};
+    if (searchString) {
+      if (!isHighlighting) {
+        setIsHighlighting(true);
+        const words = searchString
+          .toLowerCase()
+          .trim()
+          .split(' ')
+          .map((w) => w.trim());
+        const css = highlightWords(words);
+        setParagraphsCSS(css.paragraphs);
+        setSearchHighlightCSS(css.search);
+        setIsHighlighting(false);
+      } else {
+        setParagraphsCSS('');
+        setSearchString('');
+      }
+    }
```

Comment:
set highlights using `useEffect` to prevent unnecessary re-renders, this gets rid of the need for using the `onlyCallOnce` function.